tests/usb/usb.go

```diff
@@ -0,0 +1,117 @@
+	)
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		kv := util.GetCurrentKv(virtClient)
+		config = kv.Spec.Configuration
+
+		nodeName := tests.NodeNameWithHandler()
+		Expect(nodeName).ToNot(BeEmpty())
+
+		// Emulated USB devices only on c9s providers. Remove this when sig-compute 1.26 is the
+		// oldest sig-compute with test with.
+		// See: https://github.com/kubevirt/project-infra/pull/2922
+		stdout, err := tests.ExecuteCommandInVirtHandlerPod(nodeName, []string{"dmesg"})
+		Expect(err).ToNot(HaveOccurred())
+		if strings.Count(stdout, "idVendor=46f4") == 0 {
+			Skip("No emulated USB devices present for functional test.")
+		}
+	})
+
+	AfterEach(func() {
+		kv := util.GetCurrentKv(virtClient)
+		// Reinitialized the DeveloperConfiguration to avoid to influence the next test
+		config = kv.Spec.Configuration
+		config.DeveloperConfiguration = &v1.DeveloperConfiguration{}
+		config.PermittedHostDevices = &v1.PermittedHostDevices{}
+		tests.UpdateKubeVirtConfigValueAndWait(config)
+	})
+
+	Context("with usb storage", func() {
+		DescribeTable("with emulated USB devices", func(deviceNames []string) {
+			const resourceName = "kubevirt.io/usb-storage"
+
+			By("Adding the emulated USB device to the permitted host devices")
+			config.DeveloperConfiguration = &v1.DeveloperConfiguration{
+				FeatureGates: []string{virtconfig.HostDevicesGate},
+			}
+			config.PermittedHostDevices = &v1.PermittedHostDevices{
+				USB: []v1.USBHostDevice{
+					{
+						ResourceName: resourceName,
+						Selectors: []v1.USBSelector{
+							{
+								Vendor:  "46f4",
+								Product: "0001",
+							}},
+					}},
+			}
+			tests.UpdateKubeVirtConfigValueAndWait(config)
+
+			By("Creating a Fedora VMI with the usb host device")
+			hostDevs := []v1.HostDevice{}
+			for i, name := range deviceNames {
+				hostDevs = append(hostDevs, v1.HostDevice{
+					Name:       fmt.Sprintf("usb-%d-%s", i, name),
+					DeviceName: resourceName,
+				})
+			}
+
+			randomVMI := libvmi.NewCirros()
+			randomVMI.Spec.Domain.Devices.HostDevices = hostDevs
+			vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(context.Background(), randomVMI)
+			Expect(err).ToNot(HaveOccurred())
+			libwait.WaitForSuccessfulVMIStart(vmi)
+			Expect(console.LoginToCirros(vmi)).To(Succeed())
+
+			By("Making sure the usb is present inside the VMI")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: fmt.Sprintf("%s\n", cmdNumberUSBs)},
+				&expect.BExp{R: console.RetValue(fmt.Sprintf("%d", len(deviceNames)))},
+			}, 15)).To(Succeed(), "Device not found")
+
+			// Make sure to delete the VMI before ending the test otherwise a device could still be taken
+			err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Delete(context.Background(), vmi.ObjectMeta.Name, &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred(), failedDeleteVMI)
+			libwait.WaitForVirtualMachineToDisappearWithTimeout(vmi, 180)
```

Comment:
nit: you could move this to the Aftereach

Reply:
Added to the latest iteration. Thanks